catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -755,6 +754,25 @@ public SolrQuery xpathIsFuzzy(String xpath, String literal) {
         return getXPathQuery(xpath, literal, false);
     }
 
+    @Override
+    public SolrQuery propertyIsInProximityTo(String propertyName, Integer distance, String searchTerms) {
+        return getProximityQuery(propertyName, distance, searchTerms, true);
+    }
+
+    @Override
+    public SolrQuery propertyIsNotInProximityTo(String propertyName, Integer distance, String searchTerms) {
+        return getProximityQuery(propertyName, distance, searchTerms, false);
+    }
+
+    private SolrQuery getProximityQuery(String propertyName, Integer distance, String searchTerms, boolean is) {
+        String not = is ? "" : "!";
+        String searchPhrase = QUOTE + searchTerms + QUOTE + " ~" + distance;
```

Comment:
Looking at this some more, searchTerms should call the escapeSpecialChars method (which handles quote characters).  Added a check for distance